assets/javascripts/discourse/templates/chat-browse.hbs

```diff
@@ -5,15 +5,42 @@
         {{chat-channel-settings-row channel=channel categoryChannel=true}}
       {{/each}}
     </div>
+  {{/if}}
 
-  {{else if model.topicChannels}}
+  {{#if model.topicChannels}}
     <div class="topic-channels">
       {{#each model.topicChannels as |channel|}}
         {{chat-channel-settings-row channel=channel}}
       {{/each}}
     </div>
+  {{/if}}
 
-  {{else}}
-    <EmptyState @title={{i18n "chat.no_joinable_channels_title"}} @body={{i18n "chat.no_joinable_channels_body"}} />
+  {{#if noChannelsAvailable}}
+    <div class="empty-state">
+      <span class="empty-state-title">{{i18n "chat.empty_state.title"}}</span>
+      <div class="empty-state-body">
+        {{#if model.length}}
+          {{#each model as |channel|}}
+            {{chat-channel-settings-row channel=channel}}
+          {{/each}}
+        {{else}}
+          <p>{{i18n "chat.empty_state.no_public_available"}}</p>
+        {{/if}}
```

Comment:
```suggestion
        {{#each model as |channel|}}
          {{chat-channel-settings-row channel=channel}}
        {{else}}
          <p>{{i18n "chat.empty_state.no_public_available"}}</p>
        {{/each}}
```
A little known feature of `{{#each}}` is it can take an `{{else}}` for this case!

Reply:
wowowow very cool!